src/player-gui.js

```diff
@@ -61,7 +67,17 @@ class PlayerGUI extends Component {
     let uiToRender;
     if (this.props.uis.length > 0) {
       uiToRender = this.getMatchedUI(props.uis, props.state);
-      return uiToRender ? uiToRender.template(props) : this.props.uis[0].template(props);
+      // TODO sakal I think the fallback here is problematic and will lead to implementation issues - consider undefined instead
+      // TODO sakal  I removed the invokation of the component and left it to preact otherwise displayname are not relevant
+      const template = uiToRender ? uiToRender.template : this.props.uis[0].template;
+      const presetComponent = h(template, props);
```

Comment:
isn't calling `h` redundant here? is this related to the second comment? if so can;t we tell it not to mangle it? or use something else? maybe it doesn't matter if it doesn't have performance implications, does it?

regarding the fallback, never has an issue with it so can't say why it matters in hindsight.
The main issue will be for custom implementations, but keep in mind not well defined states might get stuck if you don't offer default.

Reply:
calling `h` is required to create element with props.

regarding the fallback : changing the fallback to pick the last item to fix a bug